### DISCUSSION TITLE
Conseguir vaga em grandes cidades morando no interior é possível?

### DISCUSSION THREAD

**Discussion by luizangelomedeiros** [frontendbr/forum#203] [Discussão]
Olá Pessoal, tudo certo!?
Então, sou novo no fórum e relativamente novo na área.
Trabalho há 4 anos como Web Developer e moro em Passo Fundo, interior do RS.

No fim do ano conclui minha graduação e comecei a procurar vagas de empregos em cidades/empresas maiores que pudessem me fornecer uma perspectiva de futuro na minha carreira.
Porém quando comecei a pesquisar , vi o quão desatualizado eu estava. Não conhecia varias exigências solicitadas nas vagas, como Ember, React, Bower, Backbone entre outros.
Comecei os estudos pelos mais básicos, porém na empresa aonde eu trabalho atualmente não existe a possibilidade de aplicar nenhum destes frameworks, então fico somente com conhecimento em projetos pessoais, que poucas empresas consideram como experiência.

Há uns 2 meses decidi começar a enviar currículos do mesmo jeito e por constar uma cidade do interior do RS e talvez não ter um conhecimento avançado em algumas áreas, em 90% dos casos as empresas nem sequer respondem os e-mails. Ficar em Passo Fundo não é mais uma opção, porém não consigo obter nenhuma oportunidade para migrar para outra empresa.

Gostaria de saber de vocês se ninguém passou por algo parecido ou tem alguma dica para me dar.

Um abraço

PS: As vagas que tenho tentando são principalmente para São Paulo, Florianópolis e Curitiba

**Comment by wmartins** [comment]
@luizangelomedeiros, acho que um primeiro passo é ficar atualizado com essas coisas as quais tu mesmo te julga "desatualizado". Escolhe algumas das vagas que tu quer aplicar, junta todas essas _buzzwords_ (React, Ember, ES6, PostCSS e etc.) e tenta entender o que é cada uma delas. Se tu conseguir te organizar, estuda as que tu achar mais bacana.

Acho que o grande ponto é tu estar "ligado" no que está acontecendo, tenta assinar algumas _newsletters_, participar de eventos, responder/ler dúvidas em fóruns, escrever código, escrever _blog posts_ e assim por diante. No momento que alguém vai te recrutar, vão ver que tu é um cara atualizado, aumentando as chances de te chamarem.

Outra coisa que tu pode tentar fazer é dar uma atualizada no teu currículo, de forma que tu **venda teu peixe**. As vezes a gente acha que não faz tanta coisa nos nossos trabalhos, e acaba não colocando algumas coisas feitas no currículo. Tenta colocar tudo que pareça relevante para quem tá avaliando te chamar para um papo ou não. Se tu ajudou um colega com _git_, coloca no teu currículo. Se tu fez um _refactoring_ de um código antigo ou de um código ruim, também coloca lá. A questão toda é saber te vender. Além disso, como tu tá aplicando para vagas em outras cidades, acho que vale colocar que está disposto a te mudar, isso pode aumentar as chances de te chamarem pra uma entrevista.

Junto com essa questão do currículo, eu discordo um pouco de ti quanto ao quesito "poucas empresas consideram projetos pessoais relevantes". Acho que tem bastante gente que olha para isso com bons olhos. Portanto, se tu tiver coisas no teu GitHub que mostrem o que tu sabe, já fica mais fácil avaliar se tu é um candidato bom ou não para a vaga. Por exemplo, na empresa que trabalho atualmente (e é uma empresa relativamente grande), sempre que estão pensando em contratar alguém para o nosso time, damos uma olhada no GitHub da pessoa. De lá, dá pra se ter uma noção das experiências que a pessoa tem. Avaliamos as mensagens de commit. Se usa corretamente um gerenciamento de pacotes (não commitando os `node_modules`, por exemplo). Se segue algum padrão de código. Enfim, se vai ser um bom colega de trabalho futuramente (técnicamente falando).

Acho que é mais ou menos por aí. Por fim, tu tem que ter persistência e não pode desistir. Recolhe os _feedbacks_ que te derem das entrevistas e tenta melhorar.

Boa sorte nessa jornada! :)

Ah, sobre "conseguir vaga em grandes cidades morando no interior é possível?", eu te digo que é! Morava em Arroio dos Ratos (cidade com menos de 20 mil habitantes no RS) e consegui um emprego em Porto Alegre!

**Comment by luizangelomedeiros** [comment]
@wmartins Obrigado pelas dicas, achei bem interessante suas colocações.

Acabei me expressando errado quando comentei sobre empresas considerando projetos pessoais. O que quis dizer é que em algumas vagas que pesquisei diziam "trabalhar com X diariamente á pelos 6 meses". Tenho certeza que se eu tivesse um GitHub com vários projetos interessantes e interagisse mais com a comunidade seria de grande valor. 
A grande verdade é que tenho menos conhecimento com git do que gostaria, antigamente somente tinha experiência quando usava na faculdade, mas como era usado no netbeans - pois eram projetos em java - eu tinha uma interface que fazia todos os processos para min.
Comecei a aprender justamente pois é uma coisa essencial para qualquer empresa, juntamente com Gulp e Sass. Porém como citei anteriormente, somente faço uso em projetos pessoais pois na empresa que trabalho nada disso é adotado. 

Na sua opinião é mais fácil migrar para outra empresa estando empregado ou desempregado? Pois estou pensando em sair da empresa de qualquer jeito.
Caso realmente não consiga arranjar uma outra empresa, gostaria de fazer um intercâmbio , aprimorar meu inglês e adquirir experiências novas. E quando retornar ter o diferencial do inglês.

**Comment by wmartins** [comment]
@luizangelomedeiros, sobre as vagas que pedem um tempo _X_ de experiência com tal tecnologia, tu pode aplicar de qualquer forma e na entrevista ver o porquê dessa exigência. O importante é tu conhecer a tecnologia. E se tu estiver disposto a aprender/aprimorar e não for uma exigência tão forte o tempo, tu tem mais chance de ser contratado.

Sobre o que tu falou do _git_, sigo com o mesmo "discurso", tenta aprender, passa a usar mais o GitHub (que vai te "obrigar" a usar git) e assim tu aprende mais uma tecnologia que muitas empresas usam.

Quanto a migrar estando empregado ou desempregado, a experiência que tive em todos os meus empregos foi de trocar de empresa estando empregado. Particularmente falando, acho que só vale sair de um emprego se tu for usar a maior parte desse tempo "livre" para aprimorar os teus conhecimentos. E isso somente se tu tiver condições de ficar "parado" (sem receber salário) por esse período de tempo. Como tu mesmo falou, agora que tu tá graduado, provavelmente já tem mais tempo pra estudar, então creio que "tempo" não é o problema aqui.

E sobre o "diferencial" do inglês, é realmente um diferencial, mas, creio eu, que quando somado aos conhecimentos técnicos. A não ser que tu aplique para alguma vaga em que a comunicação seja a maior parte do teu trabalho. Caso seja um trabalho onde sejam exigidos conhecimentos técnicos, o inglês vai ajudar bastante, mas não vai ser o grande diferencial.

**Comment by kazzkiq** [comment]
@luizangelomedeiros Existe um curso muito bom (e gratuito) do @willianjusten sobre Git que com certeza pode te ajudar a "domar" essa ferramenta. [Vale **muito** a pena dar uma olhada](http://willianjusten.teachable.com/courses/git-e-github-para-iniciantes/).

Além disso, uma ótima forma de aprender novas ferramentas e frameworks (e começar a participar do mundo open-source) é participando dos projetos disponíveis no [FreeCodeCamp](https://www.freecodecamp.com/). Eles são uma plataforma que tem exatamente o objetivo de espalhar conhecimento pra quem ta entrando no mundo da programação hoje (ou querendo aprender novas ferramentas), e permitem que você aprenda com pessoas experientes trabalhando em projetos reais. É uma iniciativa muito bacana.

Eu não sei exatamente qual seu nível em JavaScript, mas se você achar pertinente, seria interessante que antes de focar 100% em aprender novas tecnologias você desse uma lida nos livros/sites que passam um _overview_ sobre a linguagem em si. Por que isso é importante? Porque ao entender a linguagem a fundo, você vai conseguir entender como os frameworks funcionam de forma muito mais simples e eficiente, sem necessariamente virar dependente de nenhum deles.

Sobre esse último parágrafo, estes são alguns links que eu acho interessantes pra começar a explorar mais sobre as "entranhas" do JavaScript (e depois aplicar esse conhecimento na hora de estudar frameworks):
- [Introdução ao JavaScript](https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/A_re-introduction_to_JavaScript) (Dá uma passada completa nos conceitos básicos da linguagem);
- [JavaScript Orientado a Objetos](https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Introduction_to_Object-Oriented_JavaScript) (Dá uma passada básica sobre OO no JavaScript);
- [Coleção: You Don't Know JS](https://github.com/getify/You-Dont-Know-JS) (Comecei a ler recentemente e recomendo muito);
- [JavaScript Design Patterns](https://addyosmani.com/resources/essentialjsdesignpatterns/book/) (Possui muitas dicas e informações relevantes sobre como a linguagem funciona).

Os itens acima vão ajudar (muito) a ter uma visão macro da linguagem, o que vai te permitir aprender frameworks e ferramentas da nossa área de forma muito mais fácil depois (além de poder aplicar essa expertise em qualquer empresa que trabalhe com JavaScript no futuro, o que agrega valor à você como profissional).

**Comment by vitor-mendrone-deel** [comment]
Acho que o pessoal já deu dicas muito boas mas é sim plenamente possível! Vale até dar uma olhada na issue #204 pra se inspirar. :grin:

**Comment by luizangelomedeiros** [comment]
@wmartins valeu pelas dicas , vou acompanhar mais o fórum aqui , tenho certeza que vou aprender muito

@kazzkiq Sensacional tuas referências , vou começar pelo git , pois acredito que seja a "base" para avançar depois em outros tópicos. Quanto a minha experiência em JavaScript, acredito que me viro bem , porém uso muito ainda o Jquery. Vou começar a estudar mais em meu tempo livre , estava meio "largado" ultimamente, descansando um pouco , pois foram 5 anos sem tempo para nada  HAHAHAHA 
Muito obrigado pelas dicas 

@Mendrone valeu brother , realmente me arrependo de não ter começado a interagir antes. Vendo as issues da galera vejo que tem muita gente foda aqui, espero aprender muito ainda com vocês.

**Comment by eduardojmatos** [comment]
@luizangelomedeiros cara posso te falar que passei pelo mesmo processo.

Eu morava numa cidade do interior de SP (Amparo), e lá praticamente não existia nada de desenvolvimento web. Tive que me mudar pra Valinhos, uma cidade mais próxima de Campinas e São Paulo.

Eu senti basicamente o mesmo que você com relação a coisas pra se aprender, mas o que me minimizou isso foram os blogs e tutoriais que o pessoal sempre postava pela rede. Ler e ficar por dentro de tudo me ajudou bastante a pelo menos saber do que os outros devs falavam, mesmo não tendo experiência com as tecnologias que estavam no hype.

O fórum aqui nem se fala. É muito conteúdo e muita experiência compartilhada por aqui, vale a pena ficar algumas horas lendo heheh

**Comment by luizangelomedeiros** [comment]
@eduardojmatos Muito legal sua experiência. Realmente é importante para min ver que várias pessoas passaram o que eu estou passando e conseguiram alcançar seus objetivos.
O fórum é demais, aqui se aprende muito , tem vários profissionais de conhecimento elevado , pessoas de diferentes estados e até países contribuindo com suas experiências, é sensacional.

Muito obrigado pela resposta Eduardo , vou acompanhar mais o fórum , tentar aprender e quem sabe em breve eu possa contribuir também.

**Comment by napalmdev** [comment]
@luizangelomedeiros, eu sou uma prova viva que é possível sim morar no interior e conseguir trampo pra fora, no meu caso sou de uma cidade chamada Itaí de mais ou menos 30.000 habitantes, no interiorrrrr de SP, e tudo que eu aprendi(não é muito mas enfim) foi seguindo essa galera e os materiais encontrado pela net, e como eu disse aqui #204, graças a esse fórum eu consegui um trampo em Home Office, além é claro do conteúdo de qualidade q eles vem criando aqui já faz um tempo.

É isso aí cara, não desiste não que vai dar, só depende de você, continua nessa pegada q é impossível dar errado.

Boa sorte na sua empreitada fera.

**Comment by fakenickels** [comment]
Acho que a única grande limitação de trabalhar remoto de cidade pequena é a velocidade da conexão disponível na cidade, eu gostaria muito de trabalhar na minha cidade natal não fosse pela velocidade horrorosa da banda por lá. Ainda sonho em conseguir mexer alguns pauzinhos e trazer uma conexão boa pra lá no futuro haha, quem sabe.
Não que a banda torne impossível o seu trabalho, mas torna muito mais muuuito mais difícil.

**Comment by eduardojmatos** [comment]
Na minha época (olha o velho falando) não tinha praticamente nenhuma empresa BR com trampo remoto, o que tornava mais difícil ainda essa opção. 

@grsabreu isso é verdade, principalmente porque no remoto você precisa falar por Skype/Hangouts quase que sempre...

**Comment by ghost** [comment]
Estou numa situação semelhante, @luizangelomedeiros a diferença é que moro numa cidade do interiorrrr **pra lá** ainda, em Pernambuco, com pouco mais de 13 mil habitantes. haha E não tenho tanta experiência, comecei a estudar mesmo no início deste ano. Descobri que é o que amo fazer e estou estudando dia e noite pra conseguir um trampo em SP e ir morar lá. Até lá estou procurando alguns freelas ou Home Office pra juntar uma grana. (Parece loucura pra um iniciante, mas estou me dedicando mesmo, e sou um bom autodidata. Até agora estou me virando bem..).

A comunidade de devs é realmente muito foda! A galera se ajuda e estão sempre em busca de melhorias. Tenho uma enorme gratidão por essa galera, e espero um dia poder retribuir e contribuir com a comunidade.

**Comment by cassiocardoso** [comment]
Eu sou natural de Taubaté, SP, é uma cidade do interior, não é muito pequena, mas não possui muitas oportunidades na nossa área. Trabalhei em uma cidade vizinha, São José dos Campos até o começo de 2014. 

Nessa época eu já estava terminando a faculdade e comecei a buscar vagas em empresas de outros lugares, focando principalmente em São Paulo. Com menos de 1 mês de busca fui convidado para algumas entrevistas e logo depois entrei na empresa que estou atualmente.

Como eu não tinha a disponibilidade de me mudar imediatamente para SP, comecei com um trampo remoto até terminar a faculdade. Foi uma ótima experiência, aprendi bastante sobre o que é preciso ter para ter sucesso no trabalho remoto, e foi uma maneira de ver se eu me adaptava à empresa e deles me avaliarem também.

Então @luizangelomedeiros e @idlua continuem estudando, aprendendo e procurando por vagas que, com certeza, dá para conseguir uma oportunidade assim!
